src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2239,14 +2212,15 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
 
             // If the result type does not allow annotations, then we produce a warning because
             // the result may be null.
-            if (RequiresSafetyWarningWhenNullIntroduced(type))
-            {
-                ReportSafetyDiagnostic(ErrorCode.WRN_ConditionalAccessMayReturnNull, node.Syntax, node.Type);
-            }
 
             _currentConditionalReceiverVisitResult = default;
             _lastConditionalAccessSlot = previousConditionalAccessSlot;
             ResultType = TypeWithState.Create(type, resultState);
+            if (RequiresSafetyWarningWhenNullIntroduced(this.LvalueResultType))
```

Comment:
>this.LvalueResultType [](start = 56, length = 21)

This feels very confusing to me. The last visited node is ```node.AccessExpression``` and it is not used as an LValue. Consider adding a comment explaining the trick we do here.